install.py

```diff
@@ -5,243 +5,287 @@
-fedora = False
-mojo_lib_path_from_home = ".local/lib/mojo"
-mojo_lib_path = f"{home}/{mojo_lib_path_from_home}"
-skip_next_arg = False
-token = None
-modular = shutil.which("modular") is not None
-
-authenticated = False
-if modular:
-    authenticated = "user.id" in subprocess.run(["modular", "config-list"], capture_output=True).stdout.decode("utf-8")
-
-for arg in sys.argv:
-    if skip_next_arg:
-        skip_next_arg = False
-        continue
-
-    if arg.startswith("--dir="):
-        WORKING_DIR = arg.split("=")[1]
-    elif arg.startswith("-d="):
-        WORKING_DIR = arg.split("=")[1]
-    elif arg == "--global":
-        install_global = True
-    elif arg == "-g":
-        install_global = True
-    elif arg == "--mojo":
-        onlyMojo = True
-    elif arg == "-m":
-        onlyMojo = True
-    elif arg == "--fedora":
-        fedora = True
-    elif arg == "-f":
-        fedora = True
-    elif arg == "--modular-token":
-        index = sys.argv.index(arg) + 1
-        if index >= len(sys.argv):
-            print("No token provided")
-            exit(1)
-        token = sys.argv[index]
 
-        if token == "" or not token.startswith("mut_") or not len(token) == 36:
-            print("Invalid token")
+class Mojo(object):
+    __slots__ = ("args", "arch", "home", "working_dir", "mojo_lib_path_from_home", "mojo_lib_path",
+                 "install_global", "onlyMojo", "fedora", "skip_next_arg", "authenticated",
+                 "rc_pth", "rc_file", "token", "modular"
+                 )
+
+    def __init__(self, args: list[str], home: str | None = "~"):
+        self.args = args
+        self.arch = "x86_64-linux-gnu"
+        self.home = home if home is not None else "~"
+        self.working_dir = "~/.local/arch-mojo"
+        self.mojo_lib_path_from_home = ".local/lib/mojo"
+        self.mojo_lib_path = f"{self.home}/{self.mojo_lib_path_from_home}"
+        self.install_global = False
+        self.onlyMojo = False
+        self.fedora = False
+        self.skip_next_arg = False
+        self.authenticated = False
+        self.rc_pth = None
+        self.rc_file = None
+        self.token = ""
+        self.modular = shutil.which("modular") is not None
+        self.handle_args()
+        self.fedora_os()
+        self.install_modular()
+        self.is_authenticated()
+        self.ncurses()
+        self.install_mojo()
+        self.rc_path()
+        self.handle_rc()
+
+    def hasstr(self, target: str, original: str) -> bool:
+        if not isinstance(target, str) or not isinstance(original, str):
+            return False
+
+        elif len(target) > len(original):
+            return False
+        
+        else:
+            start = 0
+            stop = len(target)
+
+            while start < len(original):
+                if original[start:stop] == target:
+                    return True
+                start += 1
+                stop += 1
+
+            else:
+                return False
```

Comment:
What's the exact benefit of using this over pythons internal method with the `in` keyword? I assume, pythons own version is more efficient than a custom one?

Reply:
Yes, Python's `in` is faster than my custom `hasstr` and removing it would shorten the code as well.